management/src/main/java/org/ehcache/management/providers/statistics/StandardEhcacheStatistics.java

```diff
@@ -145,6 +145,10 @@
 
         if (name.equals(statisticName)) {
           SampledStatistic<Long> count = (SampledStatistic<Long>) registeredValueStatistic.getSampledStatistic();
+          if(statisticName.endsWith("Size")) {
+            // a registeredValue size ? it should go into a SizeHistory !
+            return new SizeHistory(buildHistory(count, since), MemoryUnit.B);
```

Comment:
Is there not a way to register this type information somewhere instead of having to rely on naming conventions? That seems really brittle.